### PR TITLE
NPE fixed when fetching nested toMany association of toOne association

### PR DESCRIPTION
**Pull request by encircled**
NPE fixed when fetching nested toMany association of toOne association with pagination. 

Example domain:
Person lives in a city (lazy ManyToOne)
City has streets (lazy OneToMany)

Select persons and fetch cities and it's streets:
```
Query query = entityManager.createQuery("select p from Person p");
query.setHint(QueryHints.LEFT_FETCH, "p.city.streets");
```

NPE occurs in    
`ForeignReferenceMapping#prepareNestedJoinQueryClone` line
`nestedDataResults = new ArrayList(nestedDataResults);`

because `nestedDataResults` are not found by PK because they are being removed by "see changed code".

Basically, I suppose that the root problem is `isToManyJoin` attribute of JoinManager set to `true` when there are 2 joins one toMany and one toOne. Removed code should not have been called for toOne join, but now it is.

Anyway, I was not able to simulate the problem described in comment (I can't even imagine how first and last result "may be missing their joined rows"), so I believe it would be enough to just remove it. I have not created any unit test in eclipselink project because it is a bit complicated (and I am too lazy) to set up your environment...

### CODE REVIEW

**Comment by danielkec** [comment]
:+1:

**Comment by marcelv3612** [comment]
Hi,

Please submit the bug to [Eclipse Bugzilla](https://bugs.eclipse.org/bugs/enter_bug.cgi?product=EclipseLink) .
